src/mongocrypt-ctx.c

```diff
@@ -810,8 +810,8 @@ mongocrypt_ctx_setopt_masterkey_aws (mongocrypt_ctx_t *ctx,
    BSON_ASSERT (region_len <= INT_MAX);
    bson_append_utf8 (
       &as_bson, MONGOCRYPT_STR_AND_LEN ("region"), region, region_len);
-   BSON_ASSERT (cmk_len <= INT_MAX);
    bson_append_utf8 (&as_bson, MONGOCRYPT_STR_AND_LEN ("key"), cmk, cmk_len);
+   BSON_ASSERT (cmk_len <= INT_MAX);
```

Comment:
Suggest reverting this diff. The bounds check of `cmk_len` should be above the usage in `bson_append_utf8`.